src/Sentry/Scope.cs

```diff
@@ -225,6 +225,20 @@ public ITransactionTracer? Transaction
             try
             {
                 _transaction.Value = value;
+
+                if (Options.EnableScopeSync)
+                {
+                    if (_transaction.Value != null)
+                    {
+                        // If there is a transaction set we propagate the trace to the native layer
+                        Options.ScopeObserver?.SetTrace(_transaction.Value.TraceId, _transaction.Value.SpanId);
```

Comment:
I'm concerned about calling this callback under a lock. We could create contention if this callbackcan be slow (which if it's P/Invoke it can be relatively speaking) 

But I imagine we don't have an issue concurrent access to `Transaction` right? it's at most 1 auto instrumentation and then a competing access by a user doing their manual transaction stuff?

Reply:
> But I imagine we don't have an issue concurrent access to Transaction right? it's at most 1 auto instrumentation and then a competing access by a user doing their manual transaction stuff?

Correct, that is the current worst-case scenario.